libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryService.java

```diff
@@ -287,8 +292,20 @@ private boolean locationPermissionCheck() {
       int finePermission = PermissionChecker.checkSelfPermission(MapboxTelemetry.applicationContext,
         Manifest.permission.ACCESS_FINE_LOCATION);
 
+      checkFinePermission(finePermission);
+
       return coarsePermission == PackageManager.PERMISSION_GRANTED
         || finePermission == PackageManager.PERMISSION_GRANTED;
     }
   }
+
+  private void checkFinePermission(int finePermission) {
+    if (finePermission != PackageManager.PERMISSION_GRANTED) {
+      try {
+        throw new Exception(MISSING_FINE_PERMISSION);
```

Comment:
what's the benefit of throwing exception in try and catch block? should we just return false if fine permission is unavailable? does that make sense if we're currently using passive provider which is not going to work with coarse permission?

Reply:
Yeah that was the question I was asking earlier. If we wanted to prevent the crash (with the false return here) and keep the exception to figure out why the listener is not started or just throw the exception and allow the crash to force the change. I'll do the first one. Thanks for the confirmation.